src/Init/Data/Float.lean

```diff
@@ -26,43 +26,96 @@ opaque floatSpec : FloatSpec := {
   decLe := fun _ _ => inferInstanceAs (Decidable True)
 }
 
-/-- Native floating point type, corresponding to the IEEE 754 *binary64* format
-(`double` in C or `f64` in Rust). -/
+/--
+64-bit floating-point numbers.
+
+`Float` corresponds to the IEEE 754 *binary64* format (`double` in C or `f64` in Rust).
+Floating-point numbers are a finite representation of a subset of the real numbers, and arithmetic
```

Comment:
I think it is confusing to have the sentences "Floating-point numbers are a finite representation of a subset of the real numbers" and "The floating-point numbers are not a strict subset of the real numbers" in the same docstring.

Reply:
That's a good point! I rephrased it to be more accurate, but it's a bit more verbose. I'll think about the phrasing for a bit longer and see if something more elegant comes up.